tests/admin_views/tests.py

```diff
@@ -4899,6 +4900,25 @@ def test_change_form_renders_correct_null_choice_value(self):
         response = self.client.get(reverse('admin:admin_views_choice_change', args=(choice.pk,)))
         self.assertContains(response, '<div class="readonly">No opinion</div>', html=True)
 
+    def test_readonly_foreignkey(self):
+        """
+        Test that ForeignKey renders as links for readonly fields if the target model
+        has an admin
+        """
+        model_with_admin = ModelWithAdmin.objects.create(name='model with admin')
+        model_without_admin = ModelWithoutAdmin.objects.create(name='model without admin')
+        model = ReadonlyModel.objects.create(
+            name='model1',
+            model_with_admin=model_with_admin,
+            model_without_admin=model_without_admin
+        )
+        response = self.client.get(reverse('admin:admin_views_readonlymodel_change', args=(model.pk,)))
+        link_to_book = reverse('admin:admin_views_modelwithadmin_change', args=(model_with_admin.pk,))
+        self.assertContains(response, '<div class="readonly"><a href="%s">model with admin</a></div>' % link_to_book,
+                            html=True)
+        # model without admin should be rendered as plaintext
+        self.assertContains(response, '<div class="readonly">model without admin</div>', html=True)
```

Comment:
Please add tests for `OneToOneField` with `parent_link` and admin, you can use the `Vodcast` model.

Reply:
As per our discussion above, this doesn't seem needed.